wp-content/themes/workday/inc/photo-header-template.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/**
+ * Given an attachment ID, generate an opening <header> tag with an inline style attribute
+ * The inline style="" contains the background-image information for the photo header.
+ * The closing </header> tag is assumed to be present elsewhere in the markup.
+ * @param WP_Post|int the attachment image's post or ID
+ * @return string HTML
+ */
+function workday_photo_header_tag( $post = null ) {
+	$element_id = 'photo-header-' . (string) $post;
+	$selector = '#' . $element_id;
+
+	if ( $post ) {
```

Comment:
Would it be easier to say

```php
if ( ! is_a( 'WP_Post', $post ) ) {
    $post = get_post( $post );
}
```

to ensure that $post is set, and then go on with the output here, rather than wrapping half the logic in a "if there is a post"?

Reply:
Eh; not an important question since we're the ones calling the function and making sure that its call always gets passed an ID.